proof/refine/TcbFlags_R.thy

```diff
@@ -0,0 +1,96 @@
+(*
+ * Copyright 2025, Proofcraft Pty Ltd
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ *)
+
+(* Lemmas connecting how the design and abstract specifications represent TCB flags.
+   That is, bitwise with a word in the design spec and as a set in the abstract spec.
+*)
+
+theory TcbFlags_R
+imports BaseRefine.Include Move_R
+begin
+
+lemma tcbFlagToWord_bit:
+  "\<exists>n<word_bits. tcbFlagToWord flag = bit n"
+  by (auto simp: tcbFlagToWord_def ex_nat_less_eq word_bits_conv split: tcb_flag.splits simp del: bit_def)
+
+lemma ex_tcbFlagToWord:
+  "tcbFlagMask !! n \<Longrightarrow> \<exists>flag. tcbFlagToWord flag = bit n"
+  by (auto simp: tcbFlagToWord_def tcbFlagMask_def split: tcb_flag.splits if_splits)
+
+lemma tcbFlagToWord_tcbFlagMask:
```

Comment:
tcbFlagToWord_and_tcbFlagMask ? I again forget what the naming policy is for idempotency rules... eq? id? idem? @lsf37 ?